components/logins/src/logins.udl

```diff
@@ -107,19 +82,41 @@ interface LoginsApiError {
     UnexpectedLoginsApiError(string reason);
 };
 
+[Trait, WithForeign]
+interface EncryptorDecryptor {
+    [Throws=LoginsApiError]
+    bytes encrypt(bytes cleartext, string description);
+
+    [Throws=LoginsApiError]
+    bytes decrypt(bytes ciphertext, string description);
+};
```

Comment:
What do you think about using an enum for the description?  I'm thinking there's only going to be a small number of cases that we need to distinguish and using an enum will reduce the amount of bytes going across the FFI.

Reply:
Yes, I very much like that. Maybe we can even go without an enum entirely; afaik we currently only use two strings, and one of them is for the tests. We could use a single static one, wdyt?

Reply:
IIRC, the description ends up in the error message when we report decryption failures to Sentry.

Maybe it's not needed anymore.  If it is, then it probably doesn't need to be part of the trait.  You could a wrapper function that calls decrypt/encrypt and then adds the message to the error type on failure.

Speaking of that, should `decrypt` return a `LoginsApiError` or a more specific error type?  I could see this working either way, but I wanted to raise this for discussion.

Reply:
Thats a good question. I improved error handling a little and added a `LoginsApiError::MissingKey`, which is used internally by `ManagedEncryptorDecryptor` and can also be used by the external implementation along with `LoginsApiError::IncorrectKey` - do you think thats specific enough?

Reply:
That sounds fine to me, I was mostly just curious on your thoughts.  When I considered similar ideas for other crates I wasn't sure exactly how I would handle it.

Reply:
Yes, I think less is more, the simpler the better. If there comes up a concrete need for more detailed bugs at some point, we can build them in :)